### PR TITLE
Arroyc/new yaml pipeline

### PR DESCRIPTION
**Pull request by arroyc**
Purpose of this PR is to have a sign build in our build pipeline and restructure existing build definitions accordingly.

### CODE REVIEW

File: images/build/Dockerfile [hunk 1]
```diff
@@ -282,16 +282,6 @@ RUN if [ -z "$AGENTBUILD" ]; then dotnet publish -r linux-x64 -o /opt/buildscrip
 # This stage is only when building in devops agents
 FROM main AS copybuildscriptsymbol
 COPY symbols /opt/buildscriptgen/
-COPY src/BuildScriptGenerator /usr/oryx/src/BuildScriptGenerator
```
Comment (kichalla) [inline images/build/Dockerfile]: Thanks for fixing this!

File: vsts/pipelines/_signbinary.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+steps:
+- powershell: 'Write-Host "Running Sign Oryx binaries"'
+  displayName: 'Show Build Phase'
+
+- powershell: |
+   Write-Host "##vso[task.setvariable variable=SignType]real"
+   Write-Host "##vso[task.setvariable variable=AGENTBUILD]true"
+  displayName: 'Set sign binary variable'
+  condition: and(in(variables['Build.Reason'], 'Schedule', 'Manual'), eq(variables['Build.DefinitionName'], 'Oryx-CI'))
```
Comment (gpcastro) [inline vsts/pipelines/_signbinary.yml]: we should avoid repeating these checks in every task. Add a new task that checks those things, and create a variable, e.g. "SignBinary" = "true" if they pass

File: vsts/pipelines/ci.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  displayName: Build and Publish Oryx Images
+  dependsOn: Job_2
+  condition: succeeded()
+  pool:
+    name: OryxLinux
+    demands: sh
+    steps:
+    - script: |
+        echo "##vso[task.setvariable variable=BuildBuildImages;]true"
+        echo "Hello World"
```
Comment (kichalla) [inline vsts/pipelines/ci.yml]: remove

File: vsts/pipelines/_signbinary.yml (hunk 2 above)
Comment (arroyc) [reply vsts/pipelines/_signbinary.yml]: yes I saw it … I already have a task where I'm setting signtype variable so from next task onward I will just use it

File: vsts/pipelines/_buildTemplate.yml [hunk 4]
```diff
@@ -1,4 +1,12 @@
 steps:
+- task: DownloadPipelineArtifact@0
```
Comment (gpcastro) [inline vsts/pipelines/_buildTemplate.yml]: q: why do we need this task?

File: vsts/pipelines/_signbinary.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+steps:
+- powershell: 'Write-Host "Running Sign Oryx binaries"'
+  displayName: 'Show Build Phase'
+
+- powershell: |
+   Write-Host "##vso[task.setvariable variable=SignType]real"
+   Write-Host "##vso[task.setvariable variable=AGENTBUILD]true"
```
Comment (gpcastro) [inline vsts/pipelines/_signbinary.yml]: q: why do we also set AGENTBUILD here? Isn't signType enough?

File: vsts/pipelines/_securityChecks.yml [hunk 6]
```diff
@@ -10,6 +10,12 @@ steps:
     projects: '$(Build.SourcesDirectory)\src\**\*.csproj'
     arguments: '-o $(Build.BinariesDirectory)'
     zipAfterPublish: false
+    modifyOutputPath: false
+
+- task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
```
Comment (gpcastro) [inline vsts/pipelines/_securityChecks.yml]: why are we duplicating this task here? It is already part of _buildTemplates.yaml.

File: vsts/pipelines/_signbinary.yml (hunk 5 above)
Comment (arroyc) [reply vsts/pipelines/_signbinary.yml]: yes, the way we build the build definitions initially AGENTBUILD was required, now we have enough conditions to have just one variable which is signtype, I wanted to revisit all these cleaning tasks in a separate work-item.

File: vsts/pipelines/_securityChecks.yml (hunk 6 above)
Comment (arroyc) [reply vsts/pipelines/_securityChecks.yml]: I removed it, i think during rebase it came back, I will remove it again

File: vsts/pipelines/_buildTemplate.yml (hunk 4 above)
Comment (arroyc) [reply vsts/pipelines/_buildTemplate.yml]: we need this when we do a sign build, we need to download the signed binaries and put them in right context/directory for doing a docker build for build image

**Pull request merged**